azurelinuxagent/common/cgroup.py

```diff
@@ -184,15 +181,14 @@ def _get_cpu_percent(self):
 
         return round(float(cpu_delta * self._osutil.get_processor_cores() * 100) / float(system_delta), 3)
 
-    def collect(self):
+    def get_cpu_usage(self):
```

Comment:
The implementation of collect returned an array of different length with different items for each of the controllers. This makes it difficult to use and error prone. I removed it and instead added explicit methods to get each metric.